GitHub actions update and tests (#204)
eradiate/eradiate: docs/rst/maintainer_guide.rst

.. _sec-maintainer_guide-release:

Continuous integration
----------------------

Eradiate has a continuous integration scheme built in `Github Actions <https://docs.github.com/en/actions>`_ .

> The action is configured in the ``ci.yml`` file.

Reviewer: ```suggestion
The action is configured in the ``.github/workflows/ci.yml`` file.
```

> Conda setup as well as the compilation of Mitsuba are handled by cmake. Therefore changes to the conda
> environment or the build process do not require any explicit updates to the CI configuration.

Reviewer: This is a little convoluted, I'd suggest the following (please correct me if I'm wrong):
```suggestion
As per the documented installation process, Conda environment setup is handled using 
the appropriate Makefile and Mitsuba build configuration is done using the CMake preset. 
No CI-specific build setup operations are required.
```

   This command skips the Setuptools and pip-compile update which could disrupt
   your Conda environment.

> Continuous integration

Author: Is this the location of the content, as you imagined it?

Reviewer: Yes, this preserves the logical positioning of the anchor.

----------------------

Eradiate has a continuous integration scheme built in `Github Actions <https://docs.github.com/en/actions>`_ .
The action is configured in the ``.github/workflows/ci.yml`` file.

As per the documented installation process, Conda environment setup is handled using
the appropriate Makefile and Mitsuba build configuration is done using the CMake preset.
No CI-specific build setup operations are required. However, to speed up the process, the GitHub action
uses Mamba instead of Conda to manage the Python environment.

The CI workflow uses caching for the compiled Mitsuba binaries. The cache is identified by the commit hash of the 
``mitsuba`` submodule and the file hashes of all .cpp and .h files in ``src/plugins/src``.

> Since the entire pipeline takes more than one hour to complete, it is triggered either manually or when
> a PR is set from `draft` to `ready-for-review`. 

Reviewer: ```suggestion
The pipeline is triggered automatically at every commit on the `main` branch. It is triggered
manually on pull requests by issuing a comment with the content:
```
run Eradiate CI
```
```

.. _sec-maintainer_guide-release:

> Continuous integration

Reviewer: Please move this section before the anchor, which is supposed to bring the reader to the release making instructions.

----------------------

Eradiate has a continuous integration scheme built in `Github Actions <https://docs.github.com/en/actions>`_ .
The action is configured in the ``.github/workflows/ci.yml`` file.

As per the documented installation process, Conda environment setup is handled using
the appropriate Makefile and Mitsuba build configuration is done using the CMake preset.

> No CI-specific build setup operations are required. However to speed up the process, the github action

Reviewer: ```suggestion
No CI-specific build setup operations are required. However to speed up the process, the GitHub action
```

uses Mamba instead of Conda to manage the Python environment.

The CI workflow uses caching for the compiled Mitsuba binaries. The cache is identified by the commit hash of the 

> mitsuba submodule and the file hashes of all .cpp and .h files in `src/plugins/src`.

Reviewer: ```suggestion
``mitsuba`` submodule and the file hashes of all .cpp and .h files in ``src/plugins/src``.
```

Since the entire pipeline takes more than one hour to complete, it is triggered either manually or when

> a PR is set from `draft` to `ready-for-review`. 

Reviewer: Does this then mean that the pipeline will run for every commit after marking the PR as ready for review?

Author: No, it runs once, when the PR is marked as ready. Now that I think about  it, this doesn't really work for our workflows. I'll check the different triggers again. It is pretty annoying that the comment based trigger only runs on the latest commit of the `main` branch, no matter what you do. See [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_comment-use-issue_comment)
